spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitAutoConfiguration.java

```diff
@@ -109,6 +111,11 @@ public CachingConnectionFactory rabbitConnectionFactory(
 			map.from(connection::getMode).whenNonNull().to(factory::setCacheMode);
 			map.from(connection::getSize).whenNonNull()
 					.to(factory::setConnectionCacheSize);
+			String connectionNamePrefix = properties.getConnectionNamePrefix();
+			if (StringUtils.hasText(connectionNamePrefix)) {
+				final AtomicInteger connectionNumber = new AtomicInteger();
+				factory.setConnectionNameStrategy(f -> connectionNamePrefix + "#" + connectionNumber.getAndIncrement());
```

Comment:
Rather than having an hardcoded strategy with the auto-incremented id, I'd very much prefer looking up for a `ConnectionNameStrategy` bean in the context. Does that sound sensible?